src/main/scala/memeid/Constructors.scala

```diff
@@ -42,4 +43,17 @@ trait Constructors {
       })
     }
 
+  // Construct a v4 (random) UUID.
+  def v4[F[_]: Sync]: F[UUID] = Sync[F].delay(new UUID.V4(JUUID.randomUUID))
+
+  def random[F[_]: Sync]: F[UUID] = v4
+
+  // Construct a v4 (random) UUID from the given `msb` and `lsb`.
+  def v4[F[_]: Sync](msb: Long, lsb: Long): F[UUID] = Sync[F].delay {
```

Comment:
Why creation of `v4` UUIDs from `msb` and `lsb` has to be wrapped in `delay`?

Reply:
There's no need, but I've removed the F[_] type parameter in a separate (upcoming) PR https://github.com/47deg/memeid/pull/42